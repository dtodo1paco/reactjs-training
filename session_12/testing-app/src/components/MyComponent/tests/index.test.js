import React from 'react';
import { shallow, mount } from 'enzyme';
import Component from '../index';

describe("MyComponent snapshot", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<Component />);
  });
  it("should render initial layout", () => {
    // when
    const wrapper = shallow(<Component title="aasaa" />);
    // then
    expect(wrapper.getElements()).toMatchSnapshot();
  });

});


describe("MyComponent", () => {
  it("should initialize component state", () => {
    // given
    const wrapper = shallow(<Component />);
    expect(wrapper.state('myInput')).toBeDefined();
  });

  it("should create an entry in component state", () => {
    // given
    const wrapper = shallow(<Component />);
    const form = wrapper.find('input');
    // when
    form.props().onChange({target: {
      name: 'myInput',
      value: 'myValue'
    }});
    // then
    expect(wrapper.state('myInput')).toBeDefined();
  });
  it("should create an entry in component state with the event value", () => {
    // given
    const wrapper = shallow(<Component />);
    const form = wrapper.find('input');
    // when
    form.props().onChange({target: {
      name: 'myInput',
      value: 'myValue'
    }});
    // then
    expect(wrapper.state('myInput')).toEqual('myValue');
  });

  it('render a document title', () => {
    const title = "my component is working";
    const wrapper = shallow(
      <Component title={title} />
    );
    expect(wrapper.instance().props['title']).toEqual(title);
  });
  it('render expected classNames', () => {
    const wrapper = shallow(
      <Component />
    );
    expect(wrapper.find('.myComponentClass').length).toEqual(1);
  });
});
