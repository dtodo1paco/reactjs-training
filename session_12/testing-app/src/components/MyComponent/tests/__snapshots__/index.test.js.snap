// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent snapshot should render initial layout 1`] = `
Array [
  <div
    className="myComponentClass"
  >
    Component is here 
    aasaa
    <input
      name="myInput"
      onChange={[Function]}
      type="text"
    />
  </div>,
]
`;
